libraries/rush-lib/src/cli/actions/PublishAction.ts

```diff
@@ -326,6 +318,14 @@ export class PublishAction extends BaseRushAction {
 
         this._setDependenciesBeforeCommit();
 
+        // Stage, commit, and push the changes to remote temp branch.
```

Comment:
Something we do need to be cognizant of is that the act of publishing a package might change files on disk, e.g. due to a package defining a `prepack` script. Presumably changes that result from such a script are not meant to be committed back to `main`.

Reply:
@iclanton and I have the same concern as @dmichon-msft: Historically for better or worse, Rush has NOT committed changes that occurred during publishing. In our own repo, we have CI scripts that leave Git in a dirty state, under the assumption that it will be discarded:

https://github.com/microsoft/rushstack/blob/52a1a0b341a3e801c95ab90836606a1e0ea65571/common/config/azure-pipelines/npm-publish-rush.yaml#L22-L28

So at the very least, this would be a major breaking change for `rush publish` and should maybe be controlled via an opt-in setting.

@isc30 is there another way to solve it?  I didn't read this PR thread closely, but it seems like your earlier approach was a Git command that reverts the dirty state?

Reply:
Yes, I agree that the `git add` should happen before calling the `npm publish` part.

I'm still puzzled because `_setDependenciesBeforeCommit` and `_setDependenciesBeforePublish` seem to do the same (maybe the implementation of `_setDependenciesBeforeCommit` is incorrect?). Also, these are called multiple times in a row, I'm having a hard time understanding why my fix works and the previous code doesn't.

Looking at the code, it shouldn't make a difference... will investigate further

Reply:
Okay, adding `--ignore-git-hooks` fixed the issue I was having...
After some analysis it seems like `prettier` is failing when doing the commit for some reason.

Whatever it is, its fixed by using `--ignore-git-hooks` and this PR isn't valid anymore.

Thanks a lot for your help! 😄